### ISSUE TITLE
the number of iterations of each epoch during training

### ISSUE THREAD

**Issue by ChujieXu** [csuhan/ReDet#7]
I run it using the command:

  CUDA_VISIBLE_DEVICES=1,2,3,4 ./tools/dist_train.sh configs/ReDet/ReDet_re50_refpn_1x_dota1.py 4

I used 4 gpus, and batchsize is 8.
I found that the number of iterations in each epoch is different from your log file. Mine is 1695, but yours is 1550. Why? I did not make any changes to the DOTA dataset other than run "prepare_dota1.py". Looking forward to your reply.

**Comment by csuhan** [comment]
Thank you for pointing out that! I checked it again and find:
1. Total iters are 1695 now.
2. Our 1550 comes from a rare early version using [script](https://github.com/csuhan/s2anet/blob/original_version/DOTA_devkit/prepare_dota1_ms.py) from [s2anet](https://github.com/csuhan/s2anet).

**The only difference is**:
The 1550 setting does not include the ``difficult` gts.
https://github.com/csuhan/s2anet/blob/dce96b7aeee170a44a7a2518ec9d684377da287a/DOTA_devkit/prepare_dota1_ms.py#L72-L74

While now version in this repo includes the `difficult` annotations:
https://github.com/csuhan/ReDet/blob/7430068dbffa008bf5db158d225e87de4f2bbfa5/DOTA_devkit/prepare_dota1.py#L101-L103

It means: `difficult=2` in 150 setting, while `difficult=-1` in 1695 setting.
```
def DOTA2COCOTrain(srcpath, destfile, cls_names, difficult='2', ext='.png'):
```
But I'm sure the splited data of DOTA-v1.5 is created using the script in this repo. 

I guess the difference makes minor effect to the final results.

Please infer me, if you have tested the results with the 1695 setting, thank you!

**Comment by ChujieXu** [comment]
Thank you for your answer.

I have tested both 1695 and 1550 Settings, but the accuracy of the implementation is very low (63.07% mAP in 1695 setting, 64.48% mAP in 1550 setting). I'm so confused.

I have tested on Quadro RTX 6000 with Pytorch=1.1.0, cuda=10.0.
The only change I made to the code and parameters was https://github.com/csuhan/ReDet/issues/4#issuecomment-806330053


[20210330_175152.log](https://github.com/csuhan/ReDet/files/6236945/20210330_175152.log)
[20210331_085525.log](https://github.com/csuhan/ReDet/files/6236946/20210331_085525.log)

**Comment by csuhan** [comment]
I have changed the data spliting setting to 1550.
For the performance, I suggest to:
1. Run inference using our pretrained models to validate the performace.
2. Test [RoI Transformer](https://github.com/csuhan/ReDet/blob/master/configs/DOTA/faster_rcnn_RoITrans_r50_fpn_1x_dota.py). The mAP would lies in [73.7,75.0].

**Comment by csuhan** [comment]
Hi, @ChujieXu . There is something wrong with our ReResNet pretrained models. We have updated it.
See https://github.com/csuhan/ReDet/commit/f01eb2ac0ea90bac47626feb22d42db26b0d6da4 and https://github.com/csuhan/ReDet/commit/88f8170db12a34ec342ab61571db217c9589888d

**Comment by ChujieXu** [comment]
Thank you for your reply. @csuhan  
I will try it with the updated pretrained model.

**Comment by Sairam13001** [comment]
Hello   @csuhan.

Is it same for DOTA 1.5 also? 

In DOTA_devkit/prepare_dota1_5.py 

DOTA2COCOTrain(os.path.join(dstpath, 'trainval1024'), os.path.join( 
     dstpath, 'trainval1024', 'DOTA_trainval1024.json'), wordname_15, difficult='-1') 

Could you kindly let me know if you took difficult = '2' or '-1' ?

Thank you.
